irods/client_configuration/__init__.py

```diff
@@ -99,7 +112,7 @@ def autoload(_file_to_load = None):
 
     # If there are settings to be loaded, do so.  These will override the defaults set up by the Python interpreter.
     if _file_to_load is not None:
-        load(_file = None if _file_to_load == '' else open(_file_to_load,'r'))
+        load(_file = None if _file_to_load == ''  else _file_to_load)
```

Comment:
what is this?

Reply:
why remove the 'r'?   and the whitespace is what threw me off.

Reply:
Yeah, the logic is horrible. Will work on clearing that up.

For the moment, setting the environment variable to an empty string causes `None` to be passed into load( ) as the _file argument, meaning "load from the default config-file path if it exists".  The open( ) call is now deferred til the load( ) call , so no need for the 'r' argument in that line....

Reply:
things were meant to be controlled at two different levels. load( ) and save( ) were one interface, then autoload calls into load( ) if the environment variable is defined.